src/ukify/ukify.py

```diff
@@ -197,34 +200,34 @@ def maybe_decompress(filename):
 
     if start.startswith(b'\x1f\x8b'):
         gzip = try_import('gzip')
-        return gzip.open(f).read()
+        return cast(bytes, gzip.open(f).read())
 
     if start.startswith(b'\x28\xb5\x2f\xfd'):
         zstd = try_import('zstd')
-        return zstd.uncompress(f.read())
+        return cast(bytes, zstd.uncompress(f.read()))
 
     if start.startswith(b'\x02\x21\x4c\x18'):
         lz4 = try_import('lz4.frame', 'lz4')
-        return lz4.frame.decompress(f.read())
+        return cast(bytes, lz4.frame.decompress(f.read()))
 
     if start.startswith(b'\x04\x22\x4d\x18'):
         print('Newer lz4 stream format detected! This may not boot!')
         lz4 = try_import('lz4.frame', 'lz4')
-        return lz4.frame.decompress(f.read())
+        return cast(bytes, lz4.frame.decompress(f.read()))
 
     if start.startswith(b'\x89LZO'):
         # python3-lzo is not packaged for Fedora
         raise NotImplementedError('lzo decompression not implemented')
 
     if start.startswith(b'BZh'):
         bz2 = try_import('bz2', 'bzip2')
-        return bz2.open(f).read()
+        return cast(bytes, bz2.open(f).read())
 
     if start.startswith(b'\x5d\x00\x00'):
         lzma = try_import('lzma')
-        return lzma.open(f).read()
+        return cast(bytes, lzma.open(f).read())
 
-    raise NotImplementedError(f'unknown file format (starts with {start})')
+    raise NotImplementedError(f'unknown file format (starts with {start!r})')
```

Comment:
That's a nice fix.

Reply:
Not my idea, that actually came from a ruff lint.